src/mscorlib/shared/System/IO/BinaryWriter.cs

```diff
@@ -389,6 +390,62 @@ public unsafe virtual void Write(String value)
             }
         }
 
+        public virtual void Write(ReadOnlySpan<byte> value)
+        {
+            if (GetType() == typeof(BinaryWriter))
+            {
+                OutStream.Write(value);
+            }
+            else
+            {
+                int length = value.Length;
+                byte[] buffer = ArrayPool<byte>.Shared.Rent(length);
+                try
+                {
+                    value.CopyTo(buffer);
+                    Write(buffer, 0, length);
+                }
+                finally
+                {
+                    ArrayPool<byte>.Shared.Return(buffer);
+                }
+            }
+        }
```

Comment:
The `Write(ReadOnlySpan<byte> value)` method LGTM now. (Though as a nit, I'm not sure the `length` local is necessary.)